### PR TITLE
V1 pg11 compat

### PR DESCRIPTION
**Pull request by mtuncer**
pg11 related changes. Notice that this only makes cstore_fdw work with PG11. It does not add any new features of PG11.

### CODE REVIEW

File: cstore_fdw.c [hunk 1]
```diff
@@ -1907,8 +1891,8 @@ CStoreBeginForeignScan(ForeignScanState *scanState, int executorFlags)
 	TableReadState *readState = NULL;
 	Oid foreignTableId = InvalidOid;
 	CStoreFdwOptions *cstoreFdwOptions = NULL;
-	TupleTableSlot *tupleSlot = scanState->ss.ss_ScanTupleSlot;
-	TupleDesc tupleDescriptor = tupleSlot->tts_tupleDescriptor;
+	Relation currentRelation = scanState->ss.ss_currentRelation;
+	TupleDesc tupleDescriptor = currentRelation->rd_att;
```
Comment (pykello) [inline cstore_fdw.c]: Maybe use `RelationGetDescr()`

**Comment by pykello** [comment]
I get the following warning when compiling with postgres 9.3, I'm not sure if it is related to this PR or not:

```
cstore_fdw.c:1297:1: warning: no previous prototype for function 'cstore_clean_table_resources' [-Wmissing-prototypes]
cstore_clean_table_resources(PG_FUNCTION_ARGS)
^
1 warning generated.
```

File: cstore_fdw.c (hunk 1 above)
Comment (mtuncer) [reply cstore_fdw.c]: makes sense

**Comment by mtuncer** [comment]
@pykello about the warning.

Definitions is in inside cstore_fdw.c along with other UDFs.

```c
PG_FUNCTION_INFO_V1(cstore_clean_table_resources);
```
I did not get why PG93 throw that warning.

**Comment by pykello** [comment]
I looked into this a bit. In PG93, `PG_FUNCTION_INFO_V1` doesn't include the declaration of the function. In other versions it does. In PG93, it is defined as:

```C
#define PG_FUNCTION_INFO_V1(funcname) \
extern PGDLLEXPORT const Pg_finfo_record * CppConcat(pg_finfo_,funcname)(void); \
const Pg_finfo_record * \
CppConcat(pg_finfo_,funcname) (void) \
{ \
	static const Pg_finfo_record my_finfo = { 1 }; \
	return &my_finfo; \
} \
extern int no_such_variable
```

While in 9.4 it has been changed to:

```C
#define PG_FUNCTION_INFO_V1(funcname) \
extern Datum funcname(PG_FUNCTION_ARGS); \
extern PGDLLEXPORT const Pg_finfo_record * CppConcat(pg_finfo_,funcname)(void); \
const Pg_finfo_record * \
CppConcat(pg_finfo_,funcname) (void) \
{ \
	static const Pg_finfo_record my_finfo = { 1 }; \
	return &my_finfo; \
} \
extern int no_such_variable
```

For other UDFs we have done an explicit declaration in the `cstore_fdw.h`, so we don't get warnings for them:

```c
/* cstore_fdw.h */

/* Function declarations for utility UDFs */
extern Datum cstore_table_size(PG_FUNCTION_ARGS);

/* Function declarations for foreign data wrapper */
extern Datum cstore_fdw_handler(PG_FUNCTION_ARGS);
extern Datum cstore_fdw_validator(PG_FUNCTION_ARGS);
```

If we add `cstore_clean_table_resources()` there the problem goes away.

**Pull request merged**